sass/layout/_sidebar-right.scss

```diff
@@ -15,10 +15,12 @@
 
     &.move--left {
         @include single-transition(width, 0.25s, ease-in);
+        @include single-transition(z-index, 0s, steps(1, end));
```

Comment:
So unfortunately this new line is overriding the preceding line because, well, CSS. I actually don't love 'convenience functions' like this because while they simplify some of the complexity, the abstraction hides away how CSS works behind the scenes and things don't always work as you might have expected if you aren't aware of how the function is actually applying the CSS. If you inspect the element, you will see that there are multiple `transition` selectors being applied, one for `width` and the other for `z-index` and `z-index` is winning meaning the width animation is not longer being applied.

It totally seems like this is just adding an additional transition to the existing list of defined transitions, while in reality it is actually completely replacing it because of how transitions work. 🤷‍♂️ Unfortunately, the animation is no longer applied and also, while it looks like it's working because there is no currently no animation, if there was the z-index application would actually not be working properly.

What you probably want to do here is replace both of those lines of convenience CSS functions with something like this:

```scss
.sidebar--right {
    @include translateX(100%);
    height: 100%;
    padding: 0;
    position: fixed;
    right: 0;
    width: 400px;
    z-index: 11;

    body.announcement-bar--fixed & {
        height: calc(100% - #{$announcement-bar-height});
    }

    &.move--left {
        transition:
            width 0.25ms 0s ease-in,
            z-index 0.25s 0s step-end;  
    }

    &.sidebar--right--expanded {
        @include box-shadow(0px 8px 24px alpha-color($black, .24));
        z-index: 13;
        
        &.move--left {
            transition:
                width 0.25ms 0s ease-in,
                z-index 0.25s 0s step-start;  
        }

        .sidebar-right__body {
            overflow: hidden;
        }
    }
    ...
}
```

I'm purposely not using the convenience functions here, support for  the `transition` css properly is across the board for us.

This is basically saying that when the animation starts to expand the RHS (after adding the `sidebar--right--expanded` class), the `z-index: 13` style is applied immediately (`step-start`) to put it on top of the centre at the beginning of the transition. When the RHS is collapsed, the default `z-index: 11` is not applied till the end of the animation so that it remains on top of the centre until it is no longer overlapping the centre, at which point the `z-index` is changed back to the default to layer it behind the centre.

Hopefully that make at least some sense.

Disclaimer, I have not tried this locally as my webapp is currently in an unusable state due to some in-progress activity. :P